File: scripts/loopbench/fixtures/messy-corpus/internal/dispatch/registry.go
// Package dispatch is a tiny action registry. Handlers register themselves by
// string name (usually from init funcs) and callers invoke them by name, so
// call sites are never textually adjacent to implementations.
package dispatch

import (
	"context"
	"fmt"
	"sync"
)

// HandlerFunc handles one named action with an opaque JSON payload.
type HandlerFunc func(ctx context.Context, payload []byte) error

var (
	mu       sync.RWMutex
	handlers = map[string]HandlerFunc{}
)

// Register binds an action name to a handler. Last registration wins.
func Register(action string, h HandlerFunc) {
	mu.Lock()
	defer mu.Unlock()
	handlers[action] = h
}

// Dispatch invokes the handler registered for action.
func Dispatch(ctx context.Context, action string, payload []byte) error {
	mu.RLock()
	h, ok := handlers[action]
	mu.RUnlock()
	if !ok {
		return fmt.Errorf("dispatch: no handler for action %q", action)
	}
	return h(ctx, payload)
}

File: scripts/loopbench/fixtures/messy-corpus/web/src/registry.ts
// Action registry: UI components dispatch by string name; implementations
// register themselves at import time. Mirrors the Go dispatch package.
type ActionFn = (payload: unknown) => Promise<void>;

const actions = new Map<string, ActionFn>();

export function registerAction(name: string, fn: ActionFn): void {
  actions.set(name, fn);
}

export async function dispatch(name: string, payload: unknown): Promise<void> {
  const fn = actions.get(name);
  if (!fn) {
    throw new Error(`no action registered for ${name}`);
  }
  await fn(payload);
}
